_field-types/supported-field-types/knn-vector.md

```diff
@@ -267,3 +266,220 @@ else:
 return Byte(bval)
 ```
 {% include copy.html %}
+
+## Binary k-NN vectors
+
+You can reduce memory costs by a factor of 32 by switching from float to binary vectors.
+Using binary vector indexes can lower operational costs while maintaining high recall performance, making large-scale deployment more economical and efficient.
+
+Binary format is available for the following k-NN search types:
+
+- [Approximate k-NN]({{site.url}}{{site.baseurl}}/search-plugins/knn/approximate-knn/): Supports binary vectors only for the Faiss engine with HNSW and IVF algorithms.
+- [Script score k-NN]({{site.url}}{{site.baseurl}}/search-plugins/knn/knn-score-script/): Enables the use of binary vectors in script scoring.
+- [Painless extensions]({{site.url}}{{site.baseurl}}/search-plugins/knn/painless-functions/): Allows the use of binary vectors with Painless scripting extensions.
+
+### Requirements 
+
+There are several requirements for using binary vectors in OpenSearch k-NN plugin:
+
+- The `data_type` of the binary vector index must be `binary`.
+- The `space_type` of the binary vector index must be `hamming`.
+- The `dimension` of the binary vector index must be a multiple of 8.
+- You must convert your binary data into 8-bit signed integers (`int8`) in the [-128, 127] range. For example, the binary sequence of eight bits `0, 1, 1, 0, 0, 0, 1, 1` must be converted into its equivalent byte value of `99` to be used as a binary vector input.
+
+### Example: HNSW
+
+To create a binary vector index with the Faiss engine and HNSW algorithm, send the following request:
+
+```json
+PUT /test-binary-hnsw
+{
+  "settings": {
+    "index": {
+      "knn": true
+    }
+  },
+  "mappings": {
+    "properties": {
+      "my_vector1": {
+        "type": "knn_vector",
+        "dimension": 8,
+        "data_type": "binary",
+        "method": {
+          "name": "hnsw",
+          "space_type": "hamming",
+          "engine": "faiss",
+          "parameters": {
+            "ef_construction": 128,
+            "m": 24
+          }
+        }
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+Then ingest some documents containing binary vectors:
+
+```json
+PUT _bulk?refresh=true
+{"index": {"_index": "test-binary-hnsw", "_id": "1"}}
+{"my_vector": [7], "price": 4.4}
+{"index": {"_index": "test-binary-hnsw", "_id": "2"}}
+{"my_vector": [10], "price": 14.2}
+{"index": {"_index": "test-binary-hnsw", "_id": "3"}}
+{"my_vector": [15], "price": 19.1}
+{"index": {"_index": "test-binary-hnsw", "_id": "4"}}
+{"my_vector": [99], "price": 1.2}
+{"index": {"_index": "test-binary-hnsw", "_id": "5"}}
+{"my_vector": [80], "price": 16.5}
+```
+{% include copy-curl.html %}
+
+When querying, be sure to use a binary vector:
+
+```json
+GET /test-binary-hnsw/_search
+{
+  "size": 2,
+  "query": {
+    "knn": {
+      "my_vector1": {
+        "vector": [9],
+        "k": 2
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+### Example: IVF
+
```

Comment:
Is there any other language we can use other than "IVF method", given the connotation?

Reply:
"IVF" is the algorithm method in ANN, we currently just support IVF and HNSW for binary format.

Explained at: https://github.com/opensearch-project/documentation-website/pull/7840/files#diff-7c24a2151147d0a671328b59d626a542a6ed82e3238a7f2827ff8d41e1aef49eR277

Method definition:
```
A method definition refers to the underlying configuration of the approximate k-NN algorithm you want to use.
```
https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#method-definitions